arm-service-map/2015-11-01-preview/swagger/arm-service-map.json

```diff
@@ -0,0 +1,2508 @@
+          },
+          {
+            "name": "endTime",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "format": "date-time",
+            "description": "UTC date and time specifying the end time of an interval. When not specified the service uses DateTime.UtcNow"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/MachinesSummary"
+            }
+          },
+          "default": {
+            "description": "An error occurred while processing the request. See the error.code parameter to identify the specific error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "description": "Base for all ARM-adressable resources",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Resource identifier."
+        },
+        "type": {
+          "type": "string",
+          "description": "Resource type."
+        },
+        "name": {
+          "type": "string",
+          "description": "Resource name."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Resource ETAG."
+        },
+        "kind": {
+          "type": "string",
+          "description": "Additional resource type qualifier."
+        }
+      },
+      "x-ms-external": true,
+      "x-ms-azure-resource": true,
+      "required": [
+        "id",
+        "type",
+        "name"
+      ]
+    },
+    "ResourceReference": {
+      "description": "Represents a reference to another resource.",
+      "discriminator": "kind",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Resource URI."
+        },
+        "type": {
+          "type": "string",
+          "description": "Additional resource type qualifier.  Always present when retrieving data from the service.  Optional when sending data to the service."
+        },
+        "name": {
+          "type": "string",
+          "description": "Resource name. Always present when retrieving data from the service.  Optional when sending data to the service."
+        },
+        "kind": {
+          "type": "string",
+          "description": "Specifies the sub-class of the reference."
+        }
+      },
+      "required": [
+        "id"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "ResourceOnMachineReference": {
+      "description": "Reference to a resource on a machine.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceReference"
+        }
+      ],
+      "x-ms-discriminator-value": "ref:onmachine",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
```

Comment:
You should never flatten polymorphic types. Flattening is a compile time thing. This polymorphism is at runtime. How would the deserializer know what to flatten one level up at the runtime? The server returns a polymorphic Pet as per the swagger spec. In certain cases it could be a "Cat", in some other cases it could be a "Dog". When you flatten properties they go one level up. The generated code will have to have union of flattened properties of Cat and Dog and that would be pretty bad. I hope, I conveyed my point.